queue.cpp

```diff
@@ -0,0 +1,76 @@
+#include "queue.h"
+#include "concurrentqueue.h"
+
+typedef moodycamel::ConcurrentQueue<void*> CQType, *CQPtr;
+
+#define TO_OBJECT(x) reinterpret_cast<CQPtr>((*x))
+
+extern "C" {
+
+int cq_create(CQHandle* handle)
+{
+	if (handle == nullptr) {
+		return CQ_BAD_HANDLE;
+	}
+	
+	try {
```

Comment:
Instead of `try/catch` everywhere, surely it makes more sense to disable exceptions for this source file? The queue will be more efficient too.